code/edgemedia/src/test/java/com/adobe/marketing/mobile/edge/media/internal/MediaXDMEventGeneratorTests.kt

```diff
@@ -0,0 +1,792 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.edge.media.internal
+
+import com.adobe.marketing.mobile.edge.media.MediaConstants
+import com.adobe.marketing.mobile.edge.media.internal.xdm.XDMCustomMetadata
+import com.adobe.marketing.mobile.edge.media.internal.xdm.XDMMediaCollection
+import com.adobe.marketing.mobile.edge.media.internal.xdm.XDMMediaEvent
+import com.adobe.marketing.mobile.edge.media.internal.xdm.XDMMediaEventType
+import com.adobe.marketing.mobile.edge.media.internal.xdm.XDMMediaSchema
+import com.adobe.marketing.mobile.edge.media.internal.xdm.XDMPlayerStateData
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.mockito.ArgumentCaptor
+import org.mockito.Captor
+import org.mockito.Mockito
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import java.util.Date
+
+class MediaXDMEventGeneratorTests {
+    private lateinit var mediaInfo: MediaInfo
+    private lateinit var metadata: Map<String, String>
+    private lateinit var mediaContext: MediaContext
+    private lateinit var mockEventProcessor: MediaEventProcessor
+    private lateinit var eventGenerator: MediaXDMEventGenerator
+    private var currSessionId = 0
+    private var mockTimestamp = 0L
+    private var mockPlayhead = 0L
+
+    @Captor
+    private lateinit var eventCaptor: ArgumentCaptor<XDMMediaEvent>
+
+    @Captor
+    private lateinit var sessionIdCaptor: ArgumentCaptor<String>
+    private fun <T> capture(argumentCaptor: ArgumentCaptor<T>): T = argumentCaptor.capture()
```

Comment:
nit: Add a comment here that this is needed for function calls using non-optional parameters to wrap the `capture` parameter in a non-optional type.